spec/integration/application/agent_spec.rb

```diff
@@ -326,4 +326,75 @@
       end
     end
   end
+
+  context 'multiple agents running' do
+    it "exits if an agent is already running" do
+      server.start_server do |port|
```

Comment:
since the agent doesn't run, we could skip starting a server

Reply:
addressed in https://github.com/puppetlabs/puppet/pull/8103